### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main]
```
Comment (andrewrlee) [inline .github/workflows/codeql-analysis.yml:10]: I reckon on every push/ PR might get annoying.. but will see how it goes

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['javascript']
```
Comment (petergphillips) [inline .github/workflows/codeql-analysis.yml:27]: why did you have to override this?

**Review by petergphillips — APPROVED** [review]
LGTM

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (andrewrlee) [reply .github/workflows/codeql-analysis.yml:27]: 🤷‍♂️  this file was generated by github

**Pull request merged**